core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala

```diff
@@ -1110,64 +1123,34 @@ class WorkspaceApiServiceSpec extends ApiServiceSpec {
   it should "delete an Azure workspace" in {
     withEmptyTestDatabase { dataSource: SlickDataSource =>
       withApiServicesMockitoWSMDao(dataSource) { services =>
-        // Ensure workspace has Azure context
-        when(services.workspaceManagerDAO.getWorkspace(any[UUID], any[RawlsRequestContext]))
-          .thenReturn(new WorkspaceDescription().id(UUID.randomUUID()).azureContext(new AzureContext()))
-        // Mock happy path workspaceCreate responses
-        when(
-          services.workspaceManagerDAO.createAzureWorkspaceCloudContext(any[UUID],
-                                                                        any[String],
-                                                                        any[String],
-                                                                        any[String],
-                                                                        any[RawlsRequestContext]
-          )
+        val billingProject = RawlsBillingProject(RawlsBillingProjectName("test-azure-bp"),
+                                                 CreationStatuses.Ready,
+                                                 None,
+                                                 None,
+                                                 billingProfileId = Some(UUID.randomUUID().toString)
         )
-          .thenReturn(
-            new CreateCloudContextResult().jobReport(new JobReport().id("fake_id").status(StatusEnum.SUCCEEDED))
-          )
-        when(
-          services.workspaceManagerDAO.getWorkspaceCreateCloudContextResult(any[UUID],
-                                                                            any[String],
-                                                                            any[RawlsRequestContext]
-          )
+        val wsId = UUID.randomUUID()
+        val azureWorkspace = Workspace.buildMcWorkspace(
+          namespace = "test-azure-bp",
+          name = s"test-azure-ws-${wsId}",
+          workspaceId = wsId.toString,
+          createdDate = DateTime.now,
+          lastModified = DateTime.now,
+          createdBy = "testuser@example.com",
+          attributes = Map()
         )
-          .thenReturn(
-            new CreateCloudContextResult().jobReport(new JobReport().id("fake_id").status(StatusEnum.SUCCEEDED))
-          )
-        when(services.workspaceManagerDAO.createAzureRelay(any[UUID], any[String], any[RawlsRequestContext]))
```

Comment:
Why is all of this unnecessary now?

Reply:
I felt putting the workspace in the DB directly rather than mocking the needed dependencies and then calling the POST lent itself to a simpler test.